AdminCore.Services/EventService.cs

```diff
@@ -127,17 +129,64 @@ public void UpdateEventStatus(int eventId, EventStatuses status)
       }
     }
 
-    public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, int employeeId)
+    public EventDto CreateEvent(EventDateDto dates, EventTypes eventTypes, int employeeId, int eventWorkflowId)
     {
-      CheckEventTypeAdminLevel(eventTypes, employeeId);
-      var newEvent = BuildNewEvent(employeeId, eventTypes);
+      CheckEventTypeAdminLevel(eventTypes, employeeId);   
+      var newEvent = BuildNewEvent(employeeId, eventTypes, eventWorkflowId);
       UpdateEventDates(dates, newEvent);
+      ThrowIfDaysNoticeValidationFail((int)eventTypes, newEvent.EventDates);
```

Comment:
could `newEvent.EventDates` ever return null?

Reply:
Because this method is used in the CreateEvent stack, it should always be referencing a newly created event which will always have dates to check. This could be null if the dates creation method happened to fail

Reply:
you are calling `FirstOrDefault()` on `eventDates` in `ThrowIfDaysNoticeValidationFail` without null check

Reply:
Null handling added now